client/components/ReviewsWidget.jsx

```diff
@@ -5,142 +5,221 @@ import axios from 'axios';
-          }
-      ]
-  }
+    review_id: 91790,
+    rating: 4,
+    summary: 'Quis voluptatibus sed et.',
+    recommend: true,
+    response: null,
+    body:
+      'Aut iusto quos excepturi nobis saepe exercitationem illo et veritatis. Atque dolores deserunt repudiandae. Placeat commodi perspiciatis consequuntur in. Minus et et. Aut error est accusamus corporis quo. Quae qui reprehenderit et nisi omnis illo neque cum.',
+    date: '2020-04-02T00:00:00.000Z',
+    reviewer_name: 'Nelda76',
+    helpfulness: 13,
+    photos: [
+      {
+        id: 67851,
+        url:
+          'https://images.unsplash.com/photo-1546367791-e7447b431084?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80',
+      },
+      {
+        id: 67852,
+        url:
+          'https://images.unsplash.com/photo-1520904549193-5ab0027b3fa6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80',
+      },
+    ],
+  },
 ];
 
 export class ReviewsWidget extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      filteredReviews: dummyData,
-      productReviews: dummyData,
+      filteredReviews: [],
+      productReviews: [],
       page: 1,
+      count: 5,
       numToDisplay: 2,
       sortOrder: 'relevant',
     };
   }
 
-  getReviews() {
+  // ? Do I actually need this?
+  componentDidMount() {}
 
+  componentDidUpdate(prevProps) {
+    // Temporarily just getting that data in
+    if (this.props.productId !== prevProps.productId) {
+      this.getReviews()
+        .then((reviews) => {
+          this.setState({
+            productReviews: reviews,
+            filteredReviews: reviews,
+          });
+        });
+    }
   }
 
-  filterReviews() {
+  // Input: page
+  // Output: promise that resolves to array of reviews
+  getReviews(page = 1) {
+    let requestURL = `/api/fec2/hrnyc/reviews/?product_id=${this.props.productId}&page=${page}&sort=${this.state.sortOrder}&count=5`;
+
+    return axios.get(requestURL)
+      .then(({ data }) => {
+        return data.results;
+      })
+      // ? is this the right place to error handle?
+      .catch(err => {
+        console.error(err);
+      });
+  }
+
+  // todo: Make this function async/await
+  updateReviewList() {
+    let productReviews = this.state.productReviews;
+    let page = this.state.page;
+    let filteredReviews = this.state.filteredReviews;
+
+    // while there are not enough reviews to display
+    while (filteredReviews.length < this.state.numToDisplay) {
+      // Get some more reviews
+      let newReviews = this.getReviews((page += 1));
+      // ** EXIT LOOP IF NO MORE REVIEWS **
+      if (newReviews.length === 0) {
+        break;
+        // todo: Is this a good place to make the 'Add Review' button dissapear?
+        // todo: Does anything else have to happen here?
+      }
+      // add reviews to reivew store
+      productReviews = [...productReviews, ...newReviews];
```

Comment:
Just for my own education, and referring to what we talked about this morning, does this re-assignment of productReviews alter the state directly, since productReviews refers to this.state.productReviews?

Reply:
@jimmyko360 I believe it does not.  It is true that I do assign the state array  to productReviews so if I were to push, pop or assign a value by using `productReviews[0] = review` then I would be changing the state.  But because on line 205 I'm creating a new array and then setting productReviews equal to that new array it doesn't affect the array in state.  

This code example shows the same steps
`
let arrA = ['A']
let arrB = ['B']

let newArr = [...arrA, ...arrB]

console.log(newArr === arrA) // false
console.log(newArr) // [ 'A', 'B' ]
console.log(arrA) // [ 'A' ]
console.log(arrB) // [ 'B' ]
`